src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1509,16 +1526,18 @@ static int client_parse_message(
         }
 
         r = dhcp_option_parse(message, len, dhcp_lease_parse_options, lease, &error_message);
+        if (client->bootp && r == -ENOMSG)
```

Comment:
not that it matters but i'd swap these two checks, in order to emphasize this is an error handling block. i.e. we typically have a pattern like this:

```c
r = do_something();
if (r == -EXYZ) 
      handle_exyz();
else if (r = -EABC && undersomecondition)
      handle_eabc();
else
      return log_error_errno(r, "giving up");
```

and to emphasize this error handling structure it makes sense to always place the error number check first